routes/associations.go

```diff
@@ -0,0 +1,1036 @@
+		}
+	}
+
+	// Parse TargetUserPKID from TargetUserPublicKeyBase58Check.
+	var targetUserPKID *lib.PKID
+	if requestData.TargetUserPublicKeyBase58Check != "" {
+		targetUserPKID, err = fes.getPKIDFromPublicKeyBase58Check(
+			utxoView, requestData.TargetUserPublicKeyBase58Check,
+		)
+		if err != nil {
+			return nil, errors.New("problem getting PKID for the target user")
+		}
+	}
+
+	// Parse AppPKID from AppPublicKeyBase58Check
+	var appPKID *lib.PKID
+	if requestData.AppPublicKeyBase58Check != "" {
+		appPKID, err = fes.getPKIDFromPublicKeyBase58Check(
+			utxoView, requestData.AppPublicKeyBase58Check,
+		)
+		if err != nil {
+			return nil, errors.New("problem getting PKID for the app")
+		}
+	}
+
+	// Parse Limit.
+	if queryType == AssociationQueryTypeQuery {
+		if requestData.Limit < 0 || requestData.Limit > MaxAssociationsPerQueryLimit {
+			return nil, errors.New("invalid Limit provided")
+		}
+		if requestData.Limit == 0 {
+			requestData.Limit = MaxAssociationsPerQueryLimit
+		}
+	} else if queryType == AssociationQueryTypeCount {
+		if requestData.Limit != 0 {
+			return nil, errors.New("unsupported Limit param for count operation")
+		}
+	} else {
+		return nil, errors.New("invalid query type") // This can never happen.
+	}
+
+	// Parse LastSeenAssociationID (BlockHash) from LastSeenAssociationIdHex (string).
+	var lastSeenAssociationID *lib.BlockHash
+	if queryType == AssociationQueryTypeQuery {
+		if requestData.LastSeenAssociationID != "" {
+			lastSeenAssociationIdBytes, err := hex.DecodeString(requestData.LastSeenAssociationID)
+			if err != nil {
+				return nil, errors.New("invalid LastSeenAssociationID provided")
+			}
+			lastSeenAssociationID = lib.NewBlockHash(lastSeenAssociationIdBytes)
+		}
+	} else if queryType == AssociationQueryTypeCount {
+		if requestData.LastSeenAssociationID != "" {
+			return nil, errors.New("unsupported LastSeenAssociationID param for count operation")
+		}
+	} else {
+		return nil, errors.New("invalid query type") // This can never happen.
+	}
+
+	// Validate SortDescending.
+	if queryType == AssociationQueryTypeCount && requestData.SortDescending {
+		return nil, errors.New("unsupported SortDescending param for count operation")
+	}
+
+	// Construct association query.
+	assocationQuery := &lib.UserAssociationQuery{
+		TransactorPKID:         transactorPKID,
+		TargetUserPKID:         targetUserPKID,
+		AppPKID:                appPKID,
+		AssociationType:        []byte(requestData.AssociationType),
+		AssociationTypePrefix:  []byte(requestData.AssociationTypePrefix),
+		AssociationValue:       []byte(requestData.AssociationValue),
+		AssociationValuePrefix: []byte(requestData.AssociationValuePrefix),
+		Limit:                  requestData.Limit,
+		LastSeenAssociationID:  lastSeenAssociationID,
+		SortDescending:         requestData.SortDescending,
+	}
+	return assocationQuery, nil
+}
+
+func (fes *APIServer) CreatePostAssociation(ww http.ResponseWriter, req *http.Request) {
```

Comment:
same things from user associations apply to post associations. reviewing for things specific to post associations now.